raiden/smart_contracts/ChannelManagerLibrary.sol

```diff
@@ -76,4 +81,56 @@ library ChannelManagerLibrary {
 
         return channelAddress;
     }
+
+    /// @notice deleteChannel(address) to remove a channel after it's been settled
+    /// @dev Remove channel after it's been settled
+    /// @param channelAddress (address) address of the channel to be closed
+    /// @param partner (address) address of the partner
+    function deleteChannel(Data storage self, address partner, address channelAddress) private {
+
+        address[] ourChannels = self.nodeChannels[msg.sender];
+        address[] partnerChannels = self.nodeChannels[partner];
+
+        // remove element from sender
+        for (uint i = 0; i < ourChannels.length; ++i) {
+            if (ourChannels[i] == channelAddress) {
+                ourChannels[i] = ourChannels[ourChannels.length -1];
```

Comment:
After looking at this again I realize that there are several things with the way I initially did this. For one, my tests aren't extensive enough, which is why I didn't catch these mistakes before.

Secondly, I iterate through all channels and if I hit a channel that has been settled `deleteChannel` is called, but it's always called with the same partner even though it could be a channel with another partner. So this is something I need to take care of.

And third, as pointed out by @schmir and @LefterisJP I am modifying the array that I am iterating through, which will also make things blow up at some point.

I will rethink how to implement this in a safe way. Any good suggestions are of course welcome.